### ISSUE TITLE
net-im/telegram-desktop seems impossible to build

### ISSUE THREAD

**Issue by MrPenguin07** [msva/mva-overlay#185]
Hi,

Been building (but mostly trying and failing) the various versions and 9999 of this for past couple of years from your overlay.
Had success a few times but mostly it's fail after fail.

I've removed patches, reworked and updated patches, set the commit to be the around the date when you updated the ebuild/patches (saw you updated again recently so tried/failed again), tried to maintain my own fork of tdesktop and backport the patches...

Just can't get it to work anymore.
Any advice? I can provide more thorough errors for the various patches and compile errors but I'm either missing something or this is just destined to be forever too difficult?

Thanks

**Comment by msva** [comment]
Well, actually I'm failing to build 9999 for about few months already, but had not enough time to dive into it and fix yet.
But I'll try to look for it today/tomorrow (can't promise anything, tho)

**Comment by MrPenguin07** [comment]
Thanks for responding.

I saw the patches had been updated and the previous fixed point releases removed so hoped the 9999 would build.
I am quite invested in this already with the time I've spent on it so I'd really like to hear your thoughts on where/why it's failing and know that I am willing to help if need be!

**Comment by msva** [comment]
I've bumped telegram to 5.4.5 (and also it's deps).

TL;DR, it needs:
1) newer abseil-cpp-20240116.2-r4
2) newer protobuf (tested with 27.2)
3) newer tg_owt (which in turn needs new abseil release, and also few patches I've applied through sed)
then it seems to build fine

**Comment by MrPenguin07** [comment]
> I've bumped telegram to 5.4.5 (and also it's deps).
> 
> TL;DR, it needs:
> 
>     1. newer abseil-cpp-20240116.2-r4
> 
>     2. newer protobuf (tested with 27.2)
> 
>     3. newer tg_owt (which in turn needs new abseil release, and also few patches I've applied through sed)
>        then it seems to build fine

Absolute legend!
Built 9999 without issues.

Thanks
